### ISSUE TITLE
Template validation reports parameter value not specified when it uses a keyvault reference

### ISSUE THREAD

**Issue by BillzGitHub** [microsoft/vscode-azurearmtools#609]
Hi,

This is more of a nit as I haven't been blocked by this error. In VSCode, I have a parameter for a storage prefix (which comes from the parameter file); that is then referenced in a variable which concats it with a uniqueString

**azuredeploy.json**
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "location": {
            "type": "string",
            "defaultValue": "[resourceGroup().location]",
            "metadata": {
                "description": "Location where resource(s) will be created"
            }
        },
        "storagePrefix": {
            "type": "string",
            "minLength": 3,
            "maxLength": 11
        },
        "storageSKU": {
            "type": "string",
            "defaultValue": "Standard_LRS",
            "allowedValues": [
                "Standard_LRS",
                "Standard_GRS",
                "Standard_RAGRS",
                "Standard_ZRS",
                "Premium_LRS",
                "Premium_ZRS",
                "Standard_GZRS",
                "Standard_RAGZRS"
            ]
        }
    },
    "variables": {
        "uniqueStorageName": "[concat(parameters('storagePrefix'), uniqueString(resourceGroup().id))]"
    },
    "resources": [
        {
            "type": "Microsoft.Storage/storageAccounts",
            "apiVersion": "2019-06-01",
            "name": "[variables('uniqueStorageName')]",
            "location": "[parameters('location')]",
            "sku": {
                "name": "[parameters('storageSKU')]"
            },
            "kind": "StorageV2",
            "properties": {
                "networkAcls": {
                    "bypass": "AzureServices",
                    "virtualNetworkRules": [],
                    "ipRules": [],
                    "defaultAction": "Allow"
                },
                "supportsHttpsTrafficOnly": true,
                "encryption": {
                    "services": {
                        "file": {
                            "keyType": "Account",
                            "enabled": true
                        },
                        "blob": {
                            "keyType": "Account",
                            "enabled": true
                        }
                    },
                    "keySource": "Microsoft.Storage"
                },
                "accessTier": "Hot"
            }
        }
    ]
}
```

**azuredepoy.parameters.dev.json**
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "location": {
            "value": "westus"
        },
        "storagePrefix": {
            "value": "stcmadcm"
        },
        "storageSKU": {
            "value": "Standard_LRS"
        }
    }
}
```

The deployment, both in test and prod work fine without any errors, however within VS Code I get this error in the PROBLEMS tab 
`Template validation failed: The value for the template parameter 'storagePrefix' at line '12' and column '26' is not provided. Please see https://aka.ms/arm-deploy/#parameter-file for usage details.`

I do have the latest version of VSCode installed.

Let me know if there is anything else I can provide to help with this

**Comment by StephenWeatherford** [comment]
@BillzGitHub Thanks for reporting! The ability to validate against a parameter file is brand new in 0.9.0, so glad to see you're using it.  I can't repro the problem, though.  Can you make sure that vscode is using the correct parameter file for its validation?  You can find that in the status bar when you have the template file open, and can change it by clicking (or right-click on the editor). For example:
<img width="585" alt="image" src="https://user-images.githubusercontent.com/6913354/79519997-560de180-800a-11ea-81c8-feb8f4e4c035.png">

**Comment by BillzGitHub** [comment]
Ah, thank you! It's was using the wrong file. So I switched, but now I'm running into another template validation error regarding a keyVault reference:

**azuredeploy.json**
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "administratorLoginPassword": {
            "type": "securestring",
            "metadata": {
                "description": "This is the Azure SQL Server Administrator password"
            }
        }
}
```

```json
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "administratorLoginPassword": {
            "reference": {
                "keyVault": {
                    "id": "/subscriptions/*************/resourceGroups/*******/providers/Microsoft.KeyVault/vaults/****"
                },
                "secretName": "mysecretpassword"
            }
        }
}

**Comment by neilpeterson** [comment]
Thanks @BillzGitHub I can reproduce an issue when with this configuration.

![image](https://user-images.githubusercontent.com/7844635/79535155-77ce8f00-8031-11ea-8b32-e7efb2b918a4.png)

To note, testing the template / parameters file combo against the validation api returns a good result.

```
PS > Test-AzResourceGroupDeployment -TemplateFile ./azuredeploy.json -TemplateParameterFile ./azuredeploy.parame
ters.json -ResourceGroupName cloud-shell-storage-eastus -Verbose
VERBOSE: 10:24:04 PM - Template is valid.
```

**Comment by BillzGitHub** [comment]
Thanks Neil! Yes, I can repro on my end using your outline above
![screenshot](https://user-images.githubusercontent.com/31422854/79593303-1e497d00-8090-11ea-8112-e5771c35dd5d.png)

**Comment by neilpeterson** [comment]
Cool, thanks. We will look into it.

**Comment by neilpeterson** [comment]
@BillzGitHub fix has been completed. We will try and get this into the next release but may be cutting it close. Will update here once we know.

Thanks again.

**Comment by StephenWeatherford** [comment]
@neilpeterson This is fixed by syncing up the ARM backend code?

**Comment by neilpeterson** [comment]
Jorge mentioned on yesterdays call that he had identified the issue and created the fix.

**Comment by BillzGitHub** [comment]
Thanks for your work on this!

**Comment by StephenWeatherford** [comment]
@jorgecotillo Are you sure the fix for this is in 1.0.11?  I can still repro it.

**Comment by StephenWeatherford** [comment]
(test:  testWithLanguageServer("param-with-keyvault-reference.json")

**Comment by jorgecotillo** [comment]
@StephenWeatherford 1.0.12 is the package that contains the latest updates.

**Comment by neilpeterson** [comment]
Thanks a bunch @jorgecotillo.

**Comment by StephenWeatherford** [comment]
Test: param-with-keyvault-reference.json

**Comment by BillzGitHub** [comment]
Hi @StephenWeatherford I see you've closed this issue. I'm still seeing the issue, so I'm guess the fix will go out in the next release, correct?

**Comment by StephenWeatherford** [comment]
Correct, forgot to add the "fixed in next release" label.  Probably will be out within two weeks.

**Comment by StephenWeatherford** [comment]
Fix released with 0.11.0
